pkg/scheduler/scheduler.go

```diff
@@ -239,14 +239,14 @@ var defaultSchedulerOptions = schedulerOptions{
 }
 
 // New returns a Scheduler
-func New(client clientset.Interface,
+func New(ctx context.Context,
+	client clientset.Interface,
 	informerFactory informers.SharedInformerFactory,
 	dynInformerFactory dynamicinformer.DynamicSharedInformerFactory,
 	recorderFactory profile.RecorderFactory,
-	stopCh <-chan struct{},
 	opts ...Option) (*Scheduler, error) {
 
-	stopEverything := stopCh
+	stopEverything := ctx.Done()
 	if stopEverything == nil {
```

Comment:
do we still need this check?

Reply:
Looks like we won't need this check. It has been removed.